Surface errors distinctly

wrynearson opened the pull request:
Closes https://github.com/wri/project-zeno-next/issues/18

This PR adds a toast component to handle errors, updates styling for specific errors within the chat message window, and centralizes error handling by adding `app/hooks/useErrorHandler.ts` and updating `app/store/chatStore.ts`.

<img width="648" height="420" alt="image" src="https://github.com/user-attachments/assets/82e40f45-0fcb-42c7-bb64-716289e55a00" />

<img width="568" height="90" alt="image" src="https://github.com/user-attachments/assets/c5cc74e4-a3b7-43cd-9e62-da2433876180" />

I wasn't able to reproduce all error types/situations, so I couldn't test all of the functionality covered in this PR.

_Note to reviewer_: I relied on coding assistants for this one. Please review the logic.

LanesGood commented:
Todo: check 400 errors are properly rendering toasts. Will need to mock.

LanesGood commented on app/store/chatStore.ts:
```diff
@@ -278,9 +280,37 @@ const useChatStore = create<ChatState & ChatActions>((set, get) => ({
+        });
+      } else if (
+        error instanceof Error &&
+        (error as any).status >= 400 &&
```
All of the `any` types on the errors are causing CI to fail. Should these be `string`?

wrynearson replied:
Fixed, and also fixed other build errors in https://github.com/wri/project-zeno-next/pull/42/commits/4130307edc00b1091a929a21229bb7e8911a342c

wrynearson commented:
@LanesGood I rebased and also fixed a build error stemming from https://github.com/wri/project-zeno-next/blob/feat/surface-errors-distinctly/app/hooks/useCustomAreasList.ts. 

I may have rebased incorrectly (after the rebase, the test button to trigger an error toast reappeared), so please check to make sure that more recent commits didn't regress from what you reviewed.

kamicut commented:
<img width="1509" height="842" alt="image" src="https://github.com/user-attachments/assets/8ae7d7ea-477b-4c74-bc6b-4399916c1475" />

Adding ?debug=1 will surface the toaster debug panel
